content/new-routing.md

```diff
@@ -0,0 +1,446 @@
+
+
+```
+meteor add arillo:flow-router-helpers
+```
+
+Now that you have this package, you can use helpers in your templates to display a link to a certain route. For example, in the Todos example app, our nav links look like:
+
+
+```html
+<a href="{{pathFor 'listsShow' _id=list._id}}" title="{{list.name}}"
+    class="list-todo {{activeListClass list}}">
+```
+
+### Routing programmatically
+
+In some cases you want to change routes based on user action outside of them clicking on a link. For instance, in the example app, when a user creates a new list, we want to route them to the list they just created. We do this by calling `FlowRouter.go()` once we know the id of the new list:
+
+```js
+Template.appBody.events({
+  'click .js-new-list'() {
+    const listId = Lists.methods.insert.call();
+    FlowRouter.go('listsShow', { _id: listId });
+  }
+});
+```
+
+You can also simply change part of the URL, using the `FlowRouter.setParams()` and `FlowRouter.setQueryParams()`. For instance, if we were viewing one list and wanted to go to another, we could write:
+
+```js
+FlowRouter.setParams({_id: newList._id});
+```
+
+Of course, it is more general to call `FlowRouter.go()`, so unless you are being very specific in what you are doing it's usually better to use that.
+
+### Storing data in the URL
+
+As we discussed in the introduction, the URL is really just a serialization of some part of the client-side state the user is looking at. Although parameters can only be strings, it's possible to convert any type of data to a string via serializing it.
+
+In general if you want to store arbitrary serializable data in a URL param, you can use `EJSON.stringify()` to turn it onto a string. You'll need to URL-encode the string as well to remove any characters that have meaning in a URL:
+
+```js
+FlowRouter.setQueryParams({data: encodeURIComponent(EJSON.stringify(data))});
+```
+
+You can then get the data back out of Flow Router in the opposite way (note that Flow Router unescapes the dat for you automatically):
+
+```js
+const data = EJSON.parse(FlowRouter.getQueryParam('data'));
+```
+
+## Redirecting
+
+Sometimes, your users will end up on a page that isn't the best place for them to be. Maybe the data they were looking for has moved, maybe they were on an admin panel page and logged out, or maybe they just created a new object and you want them to end up on the page for the thing they just created.
+
+Usually, we can redirect in response to a user's action by calling `FlowRouter.go()` and friends, like in our list creation example above, but if a user browses directly to a URL that doesn't exist, it's useful to know how to redirect immediately.
+
+If a URL is simply out-of-date (sometimes you can the URL scheme of an application), you can redirect inside the `action` function of the route:
```

Comment:
Is this a joke where you accidentally left out a word of the sentence where you told me I left out a word? If so, it was a good one ;)